src/QVContainer.tsx

```diff
@@ -1,87 +1,272 @@
+      modelCopy.stages = stagesCopy;
+      this.setState({ Model: modelCopy });
+    }
+
+    private handleSSlateChange(val: string, index: number): void {
+      const modelCopy = { ...this.state.Model };
+      const stagesCopy = [...this.state.Model.stages];
+      stagesCopy[index].sourceSlate = val;
+      modelCopy.stages = stagesCopy;
+      this.setState({ Model: modelCopy });
+    }
+
+    private handleSouceChange(val: string, index: number): void {
+      const modelCopy = { ...this.state.Model };
+      const stagesCopy = [...this.state.Model.stages];
+
+      if (val === "yn") {
+        stagesCopy[index].nWinners = 1;
+        stagesCopy[index].sourceInline = "Yes,No";
+        stagesCopy[index].sourceAlternates = false;
+        stagesCopy[index].sourceSlate = null;
+      }
+
+      if (val === "inline") {
+        stagesCopy[index].sourceInline = "";
+        stagesCopy[index].sourceAlternates = false;
+        stagesCopy[index].sourceSlate = null;
+      }
+
+      if (val === "slate") {
+        stagesCopy[index].sourceInline = null;
+        stagesCopy[index].sourceAlternates = false;
+        stagesCopy[index].sourceSlate = "";
+      }
+
+      if (val === "alternates") {
+        stagesCopy[index].sourceInline = null;
+        stagesCopy[index].sourceAlternates = true;
+        stagesCopy[index].sourceSlate = null;
+      }
+
+      modelCopy.stages = stagesCopy;
+      this.setState({ Model: modelCopy });
+    }
 
-    var server = sheetRef.Server;
-    // server = "https://localhost:44321";
-    server = "https://quickvote.voter-science.com";
+    private calculateSourceValue(index: number): string {
+      const stage = this.state.Model.stages[index];
 
-    var httpClient = XC.XClient.New(server, sheetRef.AuthToken, undefined);
-    var sheetClient = new QV.QVClient(httpClient, sheetRef.SheetId);
+      if (stage.sourceInline === "Yes,No") {
+        return "yn";
+      }
 
-    this.setState({
-      SheetId: sheetRef.SheetId,
-      QVClient: sheetClient
-    }, () => {
+      if (typeof stage.sourceInline === "string") {
+        return "inline";
+      }
 
+      if (typeof stage.sourceSlate === "string") {
+        return "slate";
+      }
 
-      //  $$$ catch errr
-      sheetClient.GetModel().then(model => {
-        this.setState({ Model: model });
-        /*
-                return sheetClient.PostModel(model).then( ()=> {
-                  alert("Done!");
-                });*/
-      }).catch((err) => {
+      if (stage.sourceAlternates) {
+        return "alternates";
+      }
+    }
+
+    private updateSorting(oldIndex: number, newIndex: number) {
+      const modelCopy = { ...this.state.Model };
+      const stagesCopy = [...this.state.Model.stages];
+      arrayMove(stagesCopy, oldIndex, newIndex);
+      modelCopy.stages = stagesCopy;
+      this.setState({ Model: modelCopy });
+    }
+
+    private save(): Promise<any> {
+      const server = "https://quickvote.voter-science.com";
```

Comment:
This (and all other) API call should be on QVClient.ts.   (See PostModel() ).
We may then reuse QVClient.ts in other quick-vote related plugins.